cylc/sphinx_ext/cylc_lang/autodocumenters.py

```diff
@@ -346,3 +367,314 @@ def run(self):
+
+    @staticmethod
+    def load_global_cfg(conf_path=None):
+        """Get Global Configuration metadata:
+
+        Args:
+            Path: Global conf path.
+
+        """
+        # Load Global Config:
+        if conf_path:
+            env = os.environ
+            sub = run(
+                ['cylc', 'config', '--json'],
+                capture_output=True,
+                env=env.update({'CYLC_SITE_CONF_PATH': conf_path})
+            )
+        else:
+            sub = run(['cylc', 'config', '--json'], capture_output=True)
+
+        CylcMetadataDirective.check_subproc_output(sub)
+
+        return json.loads(sub.stdout)
+
+    @staticmethod
+    def get_global_metadata(config):
+        """
+        Additional Processing:
+        * Get lists of hosts/platforms and job runner from the config.
+        * If no title is provided, use the platform/group regex as the title.
+        * If title != regex then insert text saying which regex
+          needs matching to select this platform.
+
+        Returns:
+            A dictionary in the form:
+            'platforms': {'platform regex': {..metadata..}},
+            'platform groups': {'platform regex': {..metadata..}}
+        """
+        metadata = {}
+        for section, select_from in zip(
+            ['platforms', 'platform groups'],
+            ['hosts', 'platforms']
+        ):
+            metadata[section] = config.get(section)
+            if not metadata[section]:
+                continue
+            for key in config.get(section).keys():
+                # Grab a list of hosts or platforms that this
+                # platform or group will select from:
+                select_from = (
+                    config.get(section).get(key).get('hosts')
+                    or config.get(section).get(key).get('platforms'))
+                select_from = select_from or [key]
+                metadata[section][key]['select_from'] = select_from
+
+                # Grab the job runner this platform uses:
+                if section == 'platforms':
+                    metadata[section][key]['job_runner'] = config.get(
+                        section).get(key).get('job runner', 'background')
+        return metadata
+
+    @staticmethod
+    def global_to_node(meta, src):
+        """Convert the global metadata into rst format.
+        """
+        ret = [
+            f'.. cylc:conf:: Global Config: {src}', '',
+            f'{INDENT}.. note::', '',
+            f'{INDENT * 2}Platforms and platform groups are listed'
+            ' in the order in which Cylc will check for matches to the'
+            ' ``[runtime][NAMESPACE]platform`` setting.', '']
+
+        # For platforms in platform groupts then for hosts in platforms:
+        for settings, selects in zip(
+            ['platform groups', 'platforms'],
+            ['platforms', 'hosts']
+        ):
+            if meta.get(settings, {}):
+                ret += [f'{INDENT}.. cylc:conf:: {settings}', '']
```

Comment:
The other autodocumenters in this file use the `directive()` function for writing out directives.

This adds blanklines automatically which saves you having to to it manually.

It also auto-indents the content for you which saves you tracking the indent level.